cloudinit/config/cc_resolv_conf.py

```diff
@@ -104,12 +104,16 @@ def handle(name, cfg, cloud, log, _args):
     if "resolv_conf" not in cfg:
         log.warning("manage_resolv_conf True but no parameters provided!")
 
-    template_fn = cloud.get_template_filename('resolv.conf')
-    if not template_fn:
-        log.warning("No template found, not rendering /etc/resolv.conf")
-        return
+    try:
+        cloud.distro.generate_resolv_conf(cfg, cloud, generate_resolv_conf)
+    except Exception:
+        template_fn = cloud.get_template_filename('resolv.conf')
+        if not template_fn:
+            log.warning("No template found, not rendering /etc/resolv.conf")
+            return
 
-    generate_resolv_conf(template_fn=template_fn, params=cfg["resolv_conf"])
+        generate_resolv_conf(template_fn=template_fn,
```

Comment:
Instead of this try/except, it would be nice if we could instead move the functionality of `generate_resolv_conf` into `distros/__init__.py` so we get behavior that looks more traditionally polymorphic.